applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_two_fluids_solver.py

```diff
@@ -0,0 +1,238 @@
+import KratosMultiphysics.mpi as KratosMPI                          # MPI-python interface
+
+# Check that applications were imported in the main script
+KratosMultiphysics.CheckRegisteredApplications("FluidDynamicsApplication","MetisApplication","TrilinosApplication")
+
+# Import applications
+import KratosMultiphysics.TrilinosApplication as KratosTrilinos     # MPI solvers
+import KratosMultiphysics.FluidDynamicsApplication as KratosFluid   # Fluid dynamics application
+
+# Import serial two fluid solver
+import navier_stokes_two_fluids_solver
+import trilinos_import_model_part_utility
+
+def CreateSolver(model, custom_settings):
+    return NavierStokesMPITwoFluidsSolver(model, custom_settings)
+
+class NavierStokesMPITwoFluidsSolver(navier_stokes_two_fluids_solver.NavierStokesTwoFluidsSolver):
+
+    def _ValidateSettings(self, settings):
+
+        default_settings = KratosMultiphysics.Parameters("""{
+            "solver_type": "TwoFluids",
+            "model_part_name": "",
+            "domain_size": -1,
+            "model_import_settings": {
+                "input_type": "mdpa",
+                "input_filename": "unknown_name",
+                "reorder": false
+            },
+            "distance_reading_settings"    : {
+                "import_mode"         : "from_mdpa",
+                "distance_file_name"  : "no_distance_file"
+            },
+            "maximum_iterations": 7,
+            "echo_level": 0,
+            "consider_periodic_conditions"	: false,
+            "time_order": 2,
+            "compute_reactions": false,
+            "reform_dofs_at_each_step": false,
+            "relative_velocity_tolerance": 1e-3,
+            "absolute_velocity_tolerance": 1e-5,
+            "relative_pressure_tolerance": 1e-3,
+            "absolute_pressure_tolerance": 1e-5,
+            "linear_solver_settings":   {
+                "solver_type": "AmgclMPISolver",
+                "tolerance": 1.0e-6,
+                "max_iteration": 10,
+                "scaling": false,
+                "verbosity": 0,
+                "preconditioner_type": "None",
+                "krylov_type": "cg"
+            },
+            "volume_model_part_name" : "volume_model_part",
+            "skin_parts": [""],
+            "no_skin_parts":[""],
+            "time_stepping": {
+                "automatic_time_step" : true,
+                "CFL_number"          : 1,
+                "minimum_delta_time"  : 1e-4,
+                "maximum_delta_time"  : 0.01
+            },
+            "move_mesh_flag": false,
+            "formulation": {
+                "dynamic_tau": 1.0
+            }
+        }""")
+
+        settings.ValidateAndAssignDefaults(default_settings)
+        return settings
+
+    def __init__(self, model, custom_settings):
+        # (!!!) Note: deliberately calling the constructor of the base python solver (the parent of my parent)
+        super(NavierStokesMPITwoFluidsSolver, self).__init__(model,custom_settings)
```

Comment:
Note that you are not doing what the comment says you are doing.

Reply:
Thank you very much and sorry for missing this obvious one. I remember that I did not feel confident altering this and, thus, probably left the comment.

Reply:
I think it may be worth to consider why you would want to do that. I did not check in detail, but I can tell you that the monolithic trilinos solver skips one level of constructor to avoid doing the validation of input parameters twice. I also make sure not to initialize member variables twice (self.something = ... assigned both by your class and the base class, probably to different values). If you can be sure you are not doing this kind of things, you should be fine.

Reply:
Exactly the double assignment of member variables was my concern as well - and yes, it is unfortunately the case. At the same time, the Codacy check validated `super( parentClass, self )`  as an error. Finally, I did internet research and found that the attribute values set by the parent class constructor should be over-written. I entirely understand that this is not nice or good practice but for me it solved the complaints coming from Codacy. Sorry for this pragmatically stupid thinking. Is there a way to restore the old `super( parentClass, self )`-version and still avoid the problems with Codacy?

Reply:
If your main concern is to silence codacy, you could probably fool it by having your base class constructor call a class method (move everything from the constructor to that method) and reimplement that method in the derived class (I never tried it, just a suggestion). Personally, I do not think it is worth the cost in readability, but your call. However, keep in mind that codacy is a bit over-sensitive, and there are some instances where the things it suggests are simply not practical for Kratos (I believe it focuses too much in pure python, while we have some constraints imposed by the C interface, such as having to import some applications from files that do not specifically use them).

Reply:
I agree with you in this aspect that a change (moving the content of the constructor to another function) would deteriorate the readability. Is there a way for me to merge the pull request without "silencing" Codacy by fulfilling its remarks? If there is one, I would clearly prefer it.

Reply:
Not that I know of, but you don't need to "pass" codacy to merge (we use it as a guideline, but not passing it will not block the PR, unlike travis or appveyor).

Reply:
Thank for this hint - I was not aware of the fact that it does not have the "required" label. Thus, the change of [the questionable commit](https://github.com/KratosMultiphysics/Kratos/pull/3524/commits/d92002abbf10fb6bff24293769d35956bd6587b4) was reverted and the comment was restored.